src/controllers/commands_controller.js

```diff
@@ -0,0 +1,207 @@
+ * parser to determine intent.  Then executes the intent as an action on the
+ * thing API.
+ */
+CommandsController.post('/', function (request, response) {
+
+  if(!request.body || request.body['text'] === undefined) {
+    response.status(400).send('Text element not defined');
+    return;
+  }
+
+  aiOptions.body = JSON.stringify(getAiBody(request.body['text']));
+  rp(aiOptions)
+    .then(function(aiBody) {
+      var payload = parseAIBody(aiBody);
+      if (payload.cmd == 'IOT') {
+        thingsOptions.uri = CommandsController.gatewayHref +
+          Constants.THINGS_PATH;
+        thingsOptions.headers.Authorization = 'Bearer ' +
+          CommandsController.jwt;
+
+        rp(thingsOptions)
+          .then(function(thingBody) {
+            let jsonBody = JSON.parse(thingBody);
+            let match = payload.param.toUpperCase();
+            for (var i = 0; i < jsonBody.length; i++) {
+              var obj = jsonBody[i];
+              let name = obj.name.toUpperCase();
+              if (name == match) {
+                payload.href = obj.properties.on.href;
+                break;
+              }
+            }
+            if (payload.href != '') {
+              // Returning 201 to signify that the command was mapped to an
+              // intent and matched a 'thing' in our list.  Return a response to
+              // caller with this status before the command finishes execution
+              // as the execution can take some time (e.g. blinds)
+              response.status(201).send('Command Created');
+              iotOptions.uri = CommandsController.gatewayHref + payload.href;
+              iotOptions.headers.Authorization = 'Bearer ' +
+                CommandsController.jwt;
+
+              (payload.param2 == 'on')?
+                iotOptions.body = JSON.stringify({'on': true}):
+                iotOptions.body = JSON.stringify({'on': false});
+              rp(iotOptions)
+                .then(function() {
+                  // In the future we may want to use WS to give a status of
+                  // the disposition of the command execution..
+                })
+                .catch(function(err) {
+                  // Future, give status via WS.
+                  console.log('catch inside PUT:' + err);
+                });
+            } else {
+              response.status(404).send('Thing not found');
+            }
+          })
+          .catch(function(err) {
+            console.log('error catch:' + err);
+          });
+      } else {
+        response.status(406).send('Unable to understand command');
+      }
+    })
+    .catch(function(err) {
+      response.json('error: ' + err);
+    });
+});
+
+/**
+ * Formats the body for the request to API.ai and returns the String
+ */
+function getAiBody(text) {
+  let body = {
+    'v':'20150910',
+    'query': '',
+    'lang':'en',
+    'sessionId':'b083ae95-e39a-4954-a05b-84687f50f790',
+    'timezone':'2017-07-24T10:45:52-0400'
```

Comment:
Is hardcoding this okay?

Reply:
I removed this as it's optional and we don't have anything depending on TZ ATM.